src/mongocxx/collection.hpp

```diff
@@ -75,6 +75,8 @@ class MONGOCXX_API collection {
 
    public:
 
+    collection() noexcept;
```

Comment:
I would add a comment here like the one for the default client constructor above.